integration-tests/src/tests/client/block_corruption.rs

```diff
@@ -175,17 +176,14 @@ fn check_corrupt_block(
 /// Returns `Ok(reason)` if corrupt block wasn't parsed or had changes that are not breaking by design.
 /// Returns `Err(reason)` if corrupt block was processed or correct block wasn't processed afterwards.
 fn check_process_flipped_block_fails_on_bit(
+    env: &mut TestEnv,
+    last_block_height: BlockHeight,
     corrupted_bit_idx: usize,
 ) -> Result<anyhow::Error, anyhow::Error> {
-    let epoch_length = 5000000;
-    let mut genesis = Genesis::test(vec!["test0".parse().unwrap(), "test1".parse().unwrap()], 1);
-    genesis.config.epoch_length = epoch_length;
-    let mut env = TestEnv::builder(&genesis.config).nightshade_runtimes(&genesis).build();
-
-    let mut last_block = env.clients[0].chain.get_block_by_height(0).unwrap();
+    let mut last_block = env.clients[0].chain.get_head_block().unwrap();
 
-    let mid_height = 3;
-    for h in 1..=mid_height {
+    let mid_height = last_block_height + 3;
```

Comment:
Can we get `last_block_height` from `last_block.height()`?

Reply:
Yes, it is possible indeed

I had to fix `check_corrupt_block()` to do properly process the correct block, though

Reply:
Not a biggie, only if it is convenient and straightforward